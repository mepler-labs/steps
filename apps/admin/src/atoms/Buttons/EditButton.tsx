import styled from 'styled-components';
import React from 'react';
import { Flex } from 'grid-styled';
import { grey } from 'styles/colors';
import { Link } from 'react-router-dom';

interface Props {
  component?: string;
  to?: string;
  onClick?(): void;
}

const EditButton: React.SFC<Props> = ({ component, onClick, to }) => (
  <StyledLink to={to}>
    <StyledEditButton alignItems="center" onClick={onClick}>
      <i className="material-icons">edit</i>
      Edit{component ? ` ${component}` : ''}
    </StyledEditButton>
  </StyledLink>
);

const StyledEditButton = styled(Flex)`
  color: ${grey};
  cursor: pointer;
  text-transform: uppercase;
`;

const StyledLink = styled(Link)`
  text-decoration: none;
`;

export default EditButton;
