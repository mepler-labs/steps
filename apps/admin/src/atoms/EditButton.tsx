import styled from 'styled-components';
import React from 'react';
import { Flex } from 'grid-styled';
import { grey } from 'styles/colors';

interface Props {
  component: string;
  onClick?(): void;
}

const EditButton: React.SFC<Props> = ({ component, onClick }) => (
  <StyledEditButton alignItems="center" onClick={onClick}>
    <i className="material-icons">edit</i>
    Edit {component}
  </StyledEditButton>
);

const StyledEditButton = styled(Flex)`
  color: ${grey};
  cursor: pointer;
  text-transform: uppercase;
`;

export default EditButton;
