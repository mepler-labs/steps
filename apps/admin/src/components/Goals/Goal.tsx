import Panel from 'atoms/Panel';
import React from 'react';
import EditButton from 'atoms/Buttons/EditButton';
import styled from 'styled-components';
import { pink } from 'styles/colors';
import { remCalc } from 'styles/type';
import { Flex } from 'grid-styled';

interface Props {
  text: string;
  onEdit(): void;
  className?: string;
}

const Goal: React.SFC<Props> = ({ text, onEdit, className }) => (
  <Panel className={className}>
    <Flex justifyContent="flex-end">
      <EditButton component="Goal" onClick={onEdit} />
    </Flex>
    <Flex flexDirection="column" alignItems="center">
      <StarIcon className="material-icons">star</StarIcon>
      <StyledText>{text}</StyledText>
    </Flex>
  </Panel>
);

const StarIcon = styled.i`
  color: ${pink};
  font-size: ${remCalc(50)};
`;

const StyledText = styled.div`
  text-align: center;
  font-size: ${remCalc(64)};
`;

export default Goal;
