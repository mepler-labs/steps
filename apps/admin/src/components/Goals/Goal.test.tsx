import { shallow } from 'enzyme';
import React from 'react';
import Goal from './Goal';
import EditButton from '../../atoms/EditButton';

describe('Goal.tsx', () => {
  const onEdit = jest.fn();
  it('is defined', () => {
    const wrapper = shallow(<Goal onEdit={onEdit} text="This is my goal" />);

    expect(wrapper).toBeDefined();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Goal onEdit={onEdit} text="This is my goal" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('can be edited', () => {
    const onEdit = jest.fn();
    const wrapper = shallow(<Goal text="This is my goal" onEdit={onEdit} />);
    const editButton = wrapper.find(EditButton);

    editButton.simulate('click');

    expect(onEdit).toHaveBeenCalled();
  });
});
