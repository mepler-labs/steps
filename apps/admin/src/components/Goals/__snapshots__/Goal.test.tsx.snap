// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goal.tsx renders correctly 1`] = `
<Panel>
  <Flex
    justifyContent="flex-end"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <EditButton
      component="Goal"
    />
  </Flex>
  <Flex
    alignItems="center"
    flexDirection="column"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <styled.i
      className="material-icons"
    >
      star
    </styled.i>
    <styled.div>
      This is my goal
    </styled.div>
  </Flex>
</Panel>
`;
